.github/workflows/build_test_push.yaml

```diff
@@ -88,3 +146,22 @@ jobs:
 
       - name: run functional test
         run:  bash functional_test.sh ${{ github.sha }}-server-dev
+  add-git-tags:
+    runs-on: ubuntu-latest
+    needs: functional-tests
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: '0'
+    - name: Compute new tag
+      id: compute_new_tag
+      run: |
+        current_tag=$(git for-each-ref --sort=-v:refname --format '%(refname:lstrip=2)' |  grep -E "^v[0-9]+$" | cut -c 2- | head -n1)
+        new_tag="v$(echo "$current_tag+1" | bc)"
+        echo ::set-output name=tag::$new_tag
```

Comment:
This magic-syntax (combined with actually echo-ing it to stdout) causes an _output_ to be set in github actions, which lets us use the result of this step in other steps (and I think possible other jobs and workflows? I'll evaluate..)